packages/web3-core-method/src/index.js

```diff
@@ -408,14 +410,42 @@ Method.prototype._confirmTransaction = function(defer, result, payload) {
                             receiptJSON = JSON.stringify(receipt, null, 2);
 
                             if (receipt.status === false || receipt.status === '0x0') {
-                                utils._fireError(
-                                    new Error('Transaction has been reverted by the EVM:\n' + receiptJSON),
-                                    defer.eventEmitter,
-                                    defer.reject,
-                                    null,
-                                    receipt
-                                );
+                                try {
+                                    var revertMessage = null;
+
+                                    if (method.handleRevert) {
+                                        // Get revert reason string with eth_call
+                                        revertMessage = await method.getRevertReason(
+                                            payload.params[0],
+                                            receipt.blockNumber
+                                        );
+
+                                        if (revertMessage) { // Only throw a revert error if a revert reason is existing
+                                            utils._fireError(
+                                                errors.RevertInstructionError(revertMessage.reason, revertMessage.signature),
+                                                defer.eventEmitter,
+                                                defer.reject,
+                                                payload.callback,
```

Comment:
Ah yes, what's in this callback? Why is param different than the existing handler which uses `null`?

Reply:
Oh, thanks for seeing this! Yes, the callback should not get passed here because only the ``transactionHash`` does get returned if a callback is existing.